solutions/go/ping-pong/code/app/server.go

```diff
@@ -21,5 +22,18 @@ func main() {
 
 	defer conn.Close()
 
-	conn.Write([]byte("+PONG\r\n"))
+	buf := make([]byte, 1024)
+
+	if _, err := conn.Read(buf); err != nil {
+		fmt.Println("error reading from client: ", err.Error())
+		os.Exit(1)
+	}
+
+	// Let's use a simple substring check for now. We'll implement a proper Redis Protocol parser in later stages.
+	if bytes.Contains(buf, []byte("ping")) {
```

Comment:
Step #2 instructions say :
![image](https://user-images.githubusercontent.com/1578458/200468527-91afa280-b130-4d48-b30c-1467f82108c4.png)

Is it expected for users to validate the input at this point?

Reply:
Input validation isn't expected, it's okay to just hardcode the response. The idea here was to implement a lightweight check so that it is clear to users that the response is in response to a command (which will be properly parsed in future stages).

That said, I just saw your note on case-insensitivity, and it does feel a bit weird to implement a check that is objectively wrong (it wouldn't accept `PING`). 

Maybe this should just be a comment. 

How about we change this to: 

```go
	defer conn.Close()

	buf := make([]byte, 1024)

	if _, err := conn.Read(buf); err != nil {
		fmt.Println("error reading from client: ", err.Error())
		os.Exit(1)
	}

	// Let's ignore the client's input for now and hardcode a response. 
	// We'll implement a proper Redis Protocol parser in later stages.
	conn.Write([]byte("+PONG\r\n"))
```

@marcosnils does that sound good to you?

Reply:
LGTM!

Reply:
Done!